cmd/server/cadence/server.go

```diff
@@ -222,15 +222,32 @@ func (s *server) startService() common.Daemon {
 			log.Fatalf("not able to find advanced visibility store in config: %v", advancedVisStoreKey)
 		}
 
-		params.ESConfig = advancedVisStore.ElasticSearch
+		// there are 3 circumstances:
+		// 1. advanced visibility store == elasticsearch, use ESClient and visibilityDualManager
```

Comment:
Just out of curiosity, what's 'dual' about it? Is it that it can use ES and CAAS?

Reply:
Yeah, dual means one advanced visibility datastore(ES/Pinot/OpenSearch) and one basic visibility datastore(CAAS in this case). We will always fall back to basic visibility when there is issue with advanced visibility datastore. I refactored the dual manager to use advancedVisibilityManger from esVisibilityManger so it can be reused by Pinot.